operators/tokenizer/bpetokenizer.hpp

```diff
@@ -261,6 +261,29 @@ class VocabData {
     return it->second;
   }
 
+  const std::string DecodeEncoding(int value) {
+    for (auto& it : vocab_map_) {
```

Comment:
this looks very inefficient.

Reply:
correct - best solution to this is keep a 1:1 mapping (vocab string, id) from K->V and V->K using 2 maps, but that would be storage inefficient. wanted to check in with you before adding that to confirm we want to prioritize compute?

Reply:
you may log the used token for one utterance so that you can build a small map based on the used token only.

Reply:
so this was the original goal but the issue is after we perform byte-level bpe we are only left with input ids (so we cannot get the tokens to build the map without decoding).

For example, with input "Hey Cortana", we do not know that the tokens are ["hey", "cor", "tana"] (cannot know that cortana was split as cor-tana) until we perform bbpe and decode the input ids [2189, 851, 21396]. So then we have to find the token from the input id to get the offset mapping. Does that make sense (correct me if I am wrong)?